stateproof/stateproofMessageGenerator.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package stateproof
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/crypto/merklearray"
+	"github.com/algorand/go-algorand/crypto/stateproof"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/stateproofmsg"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+var errInvalidParams = errors.New("provided parameters are invalid")
+var errOutOfBound = errors.New("request pos is out of array bounds")
+var errProvenWeightOverflow = errors.New("overflow computing provenWeight")
+
+// The Array implementation for block headers, required to build the merkle tree from them.
+//msgp:ignore blockHeadersArray
+type blockHeadersArray []bookkeeping.LightBlockHeader
```

Comment:
nit: better name `lightHeaderArray` although Array suffix in a type type is not great